packages/keyring-controller/src/KeyringController.ts

```diff
@@ -934,80 +914,7 @@ export class KeyringController extends BaseController<
    * operation completes.
    */
   async persistAllKeyrings(): Promise<boolean> {
-    return this.#withVaultLock(async () => {
-      const { encryptionKey, encryptionSalt } = this.state;
-
-      if (!this.#password && !encryptionKey) {
-        throw new Error(KeyringControllerError.MissingCredentials);
-      }
-
-      const serializedKeyrings = await Promise.all(
-        this.#keyrings.map(async (keyring) => {
-          const [type, data] = await Promise.all([
-            keyring.type,
-            keyring.serialize(),
-          ]);
-          return { type, data };
-        }),
-      );
-
-      serializedKeyrings.push(...this.#unsupportedKeyrings);
-
-      if (
-        !serializedKeyrings.some((keyring) => keyring.type === KeyringTypes.hd)
-      ) {
-        throw new Error(KeyringControllerError.NoHdKeyring);
-      }
-
-      const updatedState: Partial<KeyringControllerState> = {};
-
-      if (this.#cacheEncryptionKey) {
-        assertIsExportableKeyEncryptor(this.#encryptor);
-
-        if (encryptionKey) {
-          const key = await this.#encryptor.importKey(encryptionKey);
-          const vaultJSON = await this.#encryptor.encryptWithKey(
-            key,
-            serializedKeyrings,
-          );
-          vaultJSON.salt = encryptionSalt;
-          updatedState.vault = JSON.stringify(vaultJSON);
-        } else if (this.#password) {
-          const { vault: newVault, exportedKeyString } =
-            await this.#encryptor.encryptWithDetail(
-              this.#password,
-              serializedKeyrings,
-            );
-
-          updatedState.vault = newVault;
-          updatedState.encryptionKey = exportedKeyString;
-        }
-      } else {
-        if (typeof this.#password !== 'string') {
-          throw new TypeError(KeyringControllerError.WrongPasswordType);
-        }
-        updatedState.vault = await this.#encryptor.encrypt(
-          this.#password,
-          serializedKeyrings,
-        );
-      }
-
-      if (!updatedState.vault) {
-        throw new Error(KeyringControllerError.MissingVaultData);
-      }
-
-      const updatedKeyrings = await this.#getUpdatedKeyrings();
-      this.update((state) => {
-        state.vault = updatedState.vault;
-        state.keyrings = updatedKeyrings;
-        if (updatedState.encryptionKey) {
-          state.encryptionKey = updatedState.encryptionKey;
-          state.encryptionSalt = JSON.parse(updatedState.vault as string).salt;
-        }
-      });
-
-      return true;
-    });
+    return this.#updateVault();
```

Comment:
This will be [later](https://github.com/MetaMask/core/pull/4182/files#diff-cacd62f731d7f6947fe841f6ba3441c7a16f764b699801b0c212e0e6c15879bfR920) wrapped in a mutex, so it can't be called internally, but `persistAllKeyrings` is called all over the place; we need to move its implementation to an internal `#updateVault` method so that we can call the private method internally and leave `persistAllKeyrings` for external calls only